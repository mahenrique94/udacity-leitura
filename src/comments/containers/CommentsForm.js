import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { connect } from 'react-redux'

import { requestSave } from '../actions'

import { getComment } from '../selectors'

import CommentsFormPage from '../pages/CommentsForm'

import Comment from '../Comment'

class CommentsForm extends Component {

    static propTypes = {
        comment: PropTypes.objectOf(Comment).isRequired,
        loading: PropTypes.bool.isRequired,
        postId: PropTypes.string.isRequired,
        save: PropTypes.func.isRequired
    }

    render() {
        const { comment, loading, postId, save } = this.props
        return <CommentsFormPage comment={ comment } loading={ loading } postId={ postId } save={ save }/>
    }

}

const mapDispatchToProps = {
    save: values => requestSave(values)
}

const mapStateToProps = state => ({
    comment: getComment(state)
})

export default connect(mapStateToProps, mapDispatchToProps)(CommentsForm)
