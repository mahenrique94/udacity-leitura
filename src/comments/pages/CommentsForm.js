import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { not } from 'utils/functions'
import { validateFields } from 'utils/formValidate'

import { Col, Row } from 'reactstrap'

import Form from 'components/form/Form'
import FormData from 'components/form/FormData'
import If from 'components/If'

import { FIELDS } from '../fields'

import Comment from '../Comment'

class CommentsForm extends Component {

    static propTypes = {
        comment: PropTypes.objectOf(Comment).isRequired,
        loading: PropTypes.bool.isRequired,
        postId: PropTypes.string.isRequired,
        save: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props)

        this.form = React.createRef()
    }

    render() {
        const { comment, loading } = this.props
        const { author, body, id, timestamp } = FIELDS
        return (
            <If condition={ not(loading) } el={ null }>
                <Form
                    className="mt-5"
                    initialValues={ comment }
                    onSubmit={ this.handleSubmit }
                    ref={ this.form }
                    validate={ this.validate }
                >
                    <FormData { ...id }/>
                    <FormData { ...timestamp }/>
                    <Row>
                        <Col xs={ 12 }>
                            <FormData { ...author }/>
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={ 12 }>
                            <FormData { ...body }/>
                        </Col>
                    </Row>
                </Form>
            </If>
        )
    }

    handleSubmit = values => {
        const { postId, save } = this.props
        values.parentId = postId
        save(values)
        this.form.current.finalForm.current.form.reset()
    }

    validate = values => validateFields(values, FIELDS)

}

export default CommentsForm
