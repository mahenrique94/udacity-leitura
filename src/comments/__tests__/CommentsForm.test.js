import React from 'react'
import renderer from 'react-test-renderer'

import CommentsForm from '../pages/CommentsForm'

describe('Tests for CommentsForm page', () => {

    const REQUIRED_PROPS = {
        comment: {},
        loading: false,
        postId: 'abc123',
        save: () => console.log('Saving...')
    }

    it('Should rendering without crash', () => {
        const tree = renderer.create(
            <CommentsForm { ...REQUIRED_PROPS }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should rendering null when loading is true', () => {
        const tree = renderer.create(
            <CommentsForm { ...REQUIRED_PROPS } loading={ true }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
