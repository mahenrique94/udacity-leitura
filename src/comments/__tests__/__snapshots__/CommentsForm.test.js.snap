// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CommentsForm page Should rendering null when loading is true 1`] = `null`;

exports[`Tests for CommentsForm page Should rendering without crash 1`] = `
<form
  className="mt-5"
  onSubmit={[Function]}
>
  <div
    className="position-relative form-group"
  >
    <input
      className="form-control"
      data-cy="input-id"
      id="id"
      maxLength={999999999}
      name="id"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="hidden"
      value=""
    />
  </div>
  <div
    className="position-relative form-group"
  >
    <input
      className="form-control"
      data-cy="input-timestamp"
      id="timestamp"
      maxLength={999999999}
      name="timestamp"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="hidden"
      value=""
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="position-relative form-group"
      >
        <label
          className=""
          htmlFor="author"
        >
          Author
        </label>
        <input
          className="is-invalid form-control"
          data-cy="input-author"
          id="author"
          maxLength={60}
          name="author"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <div
          className="is-show invalid-feedback"
        >
          Fillment of this field it's required
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="position-relative form-group"
      >
        <label
          className=""
          htmlFor="body"
        >
          Content
        </label>
        <textarea
          className="is-invalid form-control"
          data-cy="input-body"
          id="body"
          maxLength={999999999}
          name="body"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          value=""
        />
        <div
          className="is-show invalid-feedback"
        >
          Fillment of this field it's required
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-3 text-right row"
  >
    <div
      className="col-12"
    >
      <button
        className="btn btn-success btn-md disabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</form>
`;
